src/store/apps/index.ts

```diff
@@ -0,0 +1,35 @@
+import {
+  createSlice,
+  createAction,
+  PayloadAction,
+} from '@reduxjs/toolkit';
+
+import { config } from '../../config';
+
+export enum SagaActionTypes {
+  UpdateRoute = 'app/saga/updateRoute',
+}
+
+const setSelectedApp = createAction<string>(SagaActionTypes.UpdateRoute);
+
+export interface AppsState {
+  selectedApp: string,
+}
+
+const initialState: AppsState = {
+  selectedApp: config.defaultApp,
+};
+
+const slice = createSlice({
+  name: 'apps',
+  initialState,
+  reducers: {
+    receive: (state, action: PayloadAction<string>) => {
+      state.selectedApp = action.payload;
```

Comment:
I'm conflicted. The zns state containing the route also seems like an appropriate place to store the selected app. It seems unlikely that we'll access the ZNS route without the app. What are your thoughts?